pkg/drivers/bess/bess.go

```diff
@@ -543,45 +545,67 @@ func (d *Driver) createPort(port *network.Port) error {
 	var err error
 	var portDriver string
 
-	if port.DPDK {
-		return errors.New("DPDK ports are not currently supported")
-	}
-
 	if port.Virtual {
-		var vportArgs *bess_pb.VPortArg
-		portDriver = VPORT
-		if port.NamesSpace == "" {
-			// If no namespace, this must belong to default namespace so just provide IF Name and IP
-			vportArgs = &bess_pb.VPortArg{
-				Ifname:  port.IfaceName,
-				IpAddrs: []string{port.IPAddr},
+		if port.DPDK {
+			portDriver = PMDPORT
+			vdevArg := &bess_pb.PMDPortArg_Vdev{
+				Vdev: fmt.Sprintf("%s,iface=%s", port.IfaceName, port.SocketPath),
+			}
+			pmdPortArg := &bess_pb.PMDPortArg{Port: vdevArg}
+			portAny, err = ptypes.MarshalAny(pmdPortArg)
+			if err != nil {
+				log.Errorf("Failure to serialize PMD port args: %v", pmdPortArg)
+				return errors.New("failed to serialize port args")
 			}
 		} else {
-			// prepend /host dir as this is what is mounted to container
-			realNsPath := path.Join("/host", port.NamesSpace)
-			vportArg := &bess_pb.VPortArg_Netns{
-				Netns: realNsPath,
+			var vportArgs *bess_pb.VPortArg
+			portDriver = VPORT
+			if port.NamesSpace == "" {
+				// If no namespace, this must belong to default namespace so just provide IF Name and IP
+				vportArgs = &bess_pb.VPortArg{
+					Ifname:  port.IfaceName,
+					IpAddrs: []string{port.IPAddr},
+				}
+			} else {
+				// prepend /host dir as this is what is mounted to container
+				realNsPath := path.Join("/host", port.NamesSpace)
+				vportArg := &bess_pb.VPortArg_Netns{
+					Netns: realNsPath,
+				}
+				vportArgs = &bess_pb.VPortArg{
+					Ifname:  port.IfaceName,
+					Cpid:    vportArg,
+					IpAddrs: []string{port.IPAddr},
+				}
 			}
-			vportArgs = &bess_pb.VPortArg{
-				Ifname:  port.IfaceName,
-				Cpid:    vportArg,
-				IpAddrs: []string{port.IPAddr},
+			portAny, err = ptypes.MarshalAny(vportArgs)
+			if err != nil {
+				log.Errorf("Failure to serialize vport args: %v", vportArgs)
+				return errors.New("failed to serialize port args")
 			}
 		}
-		portAny, err = ptypes.MarshalAny(vportArgs)
-		if err != nil {
-			log.Errorf("Failure to serialize vport args: %v", vportArgs)
-			return errors.New("failed to serialize port args")
-		}
 	} else {
-		// Must be a kernel iface, use PCAP type port
-		portDriver = PCAPPORT
-		portArg := &bess_pb.PCAPPortArg{Dev: port.IfaceName}
-		portAny, err = ptypes.MarshalAny(portArg)
-		if err != nil {
-			log.Errorf("Failure to serialize kernel port args: %v", portArg)
-			return errors.New("failed to serialize port args")
+		if port.DPDK {
+			return errors.New("DPDK physical ports are not currently supported")
 		}
+		// Must be a kernel iface, use PCAP type port
+                if port.UnixSocket {
```

Comment:
this should be under the virtual type port if statement right?

Reply:
Comment is wrong. 
Did not move the pcap comment after adding UnixSocket.
It is not virtual, so why should it be under virtual?